Stratis.Bitcoin/Notifications/BlockNotification.cs

```diff
@@ -52,22 +68,25 @@ public virtual void Notify(uint256 startHash, CancellationToken cancellationToke
 				this.Puller.SetLocation(startBlock);
 
 				// send notifications for all the following blocks
-				while (true)
-				{
+				while (!this.reSync)
```

Comment:
Next change add to the loop also the cancellation token
while(!this.reSync && !cancellationToken.IsCancellationRequested)